File: frontend/src/lib/types/health.ts
export interface ComponentHealth {
  name: string;
  status: "ok" | "degraded" | "unavailable";
  detail: string | null;
  metadata?: Record<string, unknown>;
}

export interface HealthResponse {
  status: "ok" | "degraded" | "unavailable";
  app: string;
  version: string;
  environment: string;
  components: ComponentHealth[];
}

export interface LivenessResponse {
  status: string;
}

File: app/schemas/health.py
"""Health check response schemas."""

from typing import Any, Literal

from pydantic import BaseModel, Field


class ComponentHealth(BaseModel):
    """Health status of a single dependency."""

    name: str
    status: Literal["ok", "degraded", "unavailable"]
    detail: str | None = None
    metadata: dict[str, Any] = Field(default_factory=dict)


class HealthResponse(BaseModel):
    """Application health payload."""

    status: Literal["ok", "degraded", "unavailable"] = Field(
        description="Overall application health",
    )
    app: str
    version: str
    environment: str
    components: list[ComponentHealth] = Field(default_factory=list)
